lib/Core/Result.ts

```diff
@@ -147,12 +170,24 @@ export default class Result<T = undefined> {
     if (isDefined(this.value)) return this.value as NotUndefined<T>;
 
     // If value is undefined, throw a new TerriaError using errorOverrides
-    if (typeof errorOverrides === "string") {
-      errorOverrides = { message: errorOverrides };
-    }
     throw new TerriaError({
       message: "Unhandled required Result exception",
-      ...errorOverrides
+      ...parseOverrides(errorOverrides)
     });
   }
+
+  pushErrorTo(errors: TerriaError[], errorOverrides?: TerriaErrorOverrides) {
```

Comment:
i think a lot of these utility functions can be replaced by a single function with the signature: `fold(onError: error:TerriaError => void, onValue: value => void)`